components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/LoginContextManagementUtil.java

```diff
@@ -323,14 +325,43 @@ private static boolean isStepHasMultiOption(AuthenticationContext context) {
         return stepHasMultiOption;
     }
 
-    private static boolean isValidRequest(HttpServletRequest request, AuthenticationContext context) {
+    private static boolean canHandleAuthenticator(AuthenticationContext context, String authenticators) {
+
+        List<String> authenticatorsList = new ArrayList();
+        if (authenticators != null) {
+            String[] authenticatorIdPMappings = authenticators.split(";");
+            for (String authenticatorIdPMapping : authenticatorIdPMappings) {
+                String[] authenticatorIdPMapArr = authenticatorIdPMapping.split(":");
+                authenticatorsList.add(authenticatorIdPMapArr[0]);
+            }
+        }
+
+        Map<Integer, StepConfig> stepMap = context.getSequenceConfig().getStepMap();
+        if (stepMap != null && !stepMap.isEmpty()) {
+            StepConfig stepConfig = stepMap.get(context.getCurrentStep());
+
+            for (AuthenticatorConfig authenticatorConfig : stepConfig.getAuthenticatorList()) {
+                if (!authenticatorsList.contains(authenticatorConfig.getName())) {
+                    return false;
```

Comment:
this can return false only by checking the first entry of `stepConfig.getAuthenticatorList()` right? Is that expected?

Reply:
yes, while loading a UI on the authentication flow, the authenticators query param contains all the supported authenticators in the particular step. Here we are validating whether that matches with the step we have in the current step in context. That is done because it is not possible predict which option user will pick.